source/extensions/filters/http/common/aws/signer_impl.cc

```diff
@@ -0,0 +1,256 @@
+#include "extensions/filters/http/common/aws/signer_impl.h"
+
+#include "envoy/common/exception.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/hex.h"
+#include "common/common/stack_array.h"
+#include "common/http/headers.h"
+
+#include "openssl/evp.h"
+#include "openssl/hmac.h"
+#include "openssl/sha.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Aws {
+
+static const std::string AWS4{"AWS4"};
+static const std::string AWS4_HMAC_SHA256{"AWS4-HMAC-SHA256"};
+static const std::string AWS4_REQUEST{"aws4_request"};
+static const std::string CREDENTIAL{"Credential"};
+static const std::string SIGNED_HEADERS{"SignedHeaders"};
+static const std::string SIGNATURE{"Signature"};
+static const std::string HASHED_EMPTY_STRING{
+    "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"};
+
+DateFormatter SignerImpl::LONG_DATE_FORMATTER("%Y%m%dT%H%M00Z");
+DateFormatter SignerImpl::SHORT_DATE_FORMATTER("%Y%m%d");
+const Http::LowerCaseString SignerImpl::X_AMZ_SECURITY_TOKEN{"x-amz-security-token"};
+const Http::LowerCaseString SignerImpl::X_AMZ_DATE{"x-amz-date"};
+const Http::LowerCaseString SignerImpl::X_AMZ_CONTENT_SHA256{"x-amz-content-sha256"};
+
+void SignerImpl::sign(Http::Message& message) const {
+  const auto& credentials = credentials_provider_->getCredentials();
+  if (!credentials.accessKeyId() || !credentials.secretAccessKey()) {
+    return;
```

Comment:
Silent failure?

Reply:
Not exactly. We treat no credentials as being an anonymous request. This is important for testing situations where a developer may not want to (or is able to) authenticate it. I could log here that the request is not being signed..

Reply:
Got it, I think maybe just a comment then to explain this would be helpful.